tests/test_socket.py

```diff
@@ -23,8 +30,8 @@ def test_socket_enabled_by_default(testdir):
         import socket
 
         def test_socket():
-            socket.socket(socket.AF_INET, socket.SOCK_STREAM)
```

Comment:
by changing the test conditions, we would effectively break a promise of existing functionality, in the event that people are using socket.socket

Reply:
yea, but you can't have host checking just based on socket.socket (no host specified until connect), so we would have to put it in a completely separate flag/code-path, which we can do, but it will complicate the api for people.